components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/TenantAwareAxis2ConfigurationContextObserver.java

```diff
@@ -0,0 +1,191 @@
+import org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceFile;
+import org.wso2.carbon.identity.tenant.resource.manager.constants.TenantResourceConstants;
+import org.wso2.carbon.identity.tenant.resource.manager.exception.TenantResourceManagementException;
+import org.wso2.carbon.identity.tenant.resource.manager.internal.TenantResourceManagerDataHolder;
+import org.wso2.carbon.utils.AbstractAxis2ConfigurationContextObserver;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import java.util.List;
+
+import static org.wso2.carbon.identity.tenant.resource.manager.constants.TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_CREATING_TENANT_EVENT_PUBLISHER_CONFIGURATION_BY_CONFIG_STORE;
+import static org.wso2.carbon.identity.tenant.resource.manager.constants.TenantResourceConstants.ErrorMessages.ERROR_CODE_ERROR_WHEN_CREATING_TENANT_EVENT_PUBLISHER_CONFIGURATION_USING_SUPER_TENANT_CONFIG;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.getResourceFile;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.getResourceManager;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.handleServerException;
+
+/**
+ * Axis2Observer for generating tenant wise publisher configurations.
+ */
+public class TenantAwareAxis2ConfigurationContextObserver extends AbstractAxis2ConfigurationContextObserver {
+
+    private static final Log log = LogFactory.getLog(TenantAwareAxis2ConfigurationContextObserver.class);
+
+    /**
+     * Add the tenant wise publisher and stream Configuration in tenant loading.
+     *
+     * @param tenantId tenant ID.
+     */
+    @Override
+    public void creatingConfigurationContext(int tenantId) {
+
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        log.info("Creating configuration context for tenant domain: " + tenantDomain);
+        List<EventPublisherConfiguration> activeEventPublisherConfigurations;
+        List<EventStreamConfiguration> eventStreamConfigurationList;
+        try {
+            startSuperTenantFlow();
+            activeEventPublisherConfigurations = getSuperTenantEventPublisherConfigurations();
+            eventStreamConfigurationList = getSuperTenantEventStreamConfigurations();
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+        startTenantFlow(tenantId, tenantDomain);
+
+        createTenantEventStreamConfiguration(eventStreamConfigurationList);
+
+        if (activeEventPublisherConfigurations != null) {
+            createTenantEventPublisherConfiguration(activeEventPublisherConfigurations);
+        }
```

Comment:
Better to move this block to a separate method for readability

Reply:
moved